packages/protocol/test/common/multisig.ts

```diff
@@ -0,0 +1,382 @@
+          from: accounts[3],
+        })
+      )
+    })
+  })
+
+  describe('#replaceOwner()', () => {
+    it('should allow an existing owner to be replaced by a new one via the MultiSig', async () => {
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const txData = multiSig.contract.methods.replaceOwner(accounts[1], accounts[2]).encodeABI()
+      const tx = await multiSig.submitTransaction(multiSig.address, 0, txData, {
+        from: accounts[0],
+      })
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const txEvent = _.find(tx.logs, {
+        event: 'Confirmation',
+      })
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const txId = txEvent.args.transactionId
+      // @ts-ignore: TODO(mcortesi): fix typings
+      await multiSig.confirmTransaction(txId, { from: accounts[1] })
+      assert.isTrue(await multiSig.isOwner(accounts[2]))
+      assert.isFalse(await multiSig.isOwner(accounts[1]))
+    })
+
+    it('should not allow an external account to replace an owner', async () => {
+      // @ts-ignore
+      const txData = multiSig.contract.methods.replaceOwner(accounts[1], accounts[2]).encodeABI()
+      // @ts-ignore: TODO(mcortesi): fix typings
+      await assertRevert(
+        multiSig.submitTransaction(multiSig.address, 0, txData, {
+          from: accounts[3],
+        })
+      )
+    })
+
+    it('should not allow an owner to be replaced by the null address', async () => {
+      // @ts-ignore
+      const txData = multiSig.contract.methods.replaceOwner(accounts[1], NULL_ADDRESS).encodeABI()
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const tx = await multiSig.submitTransaction(multiSig.address, 0, txData, {
+        from: accounts[0],
+      })
+
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const txEvent = _.find(tx.logs, {
+        event: 'Confirmation',
+      })
+      const txId = txEvent.args.transactionId
+      // @ts-ignore: TODO(mcortesi): fix typings
+      await assertRevert(multiSig.confirmTransaction(txId, { from: accounts[1] }))
+    })
+  })
+
+  describe('#changeRequirement()', () => {
+    it('should allow the requirement to be changed via the MultiSig', async () => {
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const txData = multiSig.contract.methods.changeRequirement(1).encodeABI()
+      const tx = await multiSig.submitTransaction(multiSig.address, 0, txData, {
+        from: accounts[0],
+      })
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const txEvent = _.find(tx.logs, {
+        event: 'Confirmation',
+      })
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const txId = txEvent.args.transactionId
+
+      // @ts-ignore: TODO(mcortesi): fix typings
+      await multiSig.confirmTransaction(txId, { from: accounts[1] })
+      assert.equal((await multiSig.required()).toNumber(), 1)
+    })
+
+    it('should not allow an external account to change the requirement', async () => {
```

Comment:
Ditto, should this call `changeRequirement` directly?

Reply:
changed